File: app/src/main/java/com/bmn/bookfinder/adapters/FirstUseViewPagerAdapter.java
package com.bmn.bookfinder.adapters;

import android.content.Context;
import androidx.viewpager.widget.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.bmn.bookfinder.R;

import org.jetbrains.annotations.NotNull;

public class FirstUseViewPagerAdapter extends PagerAdapter {

    private LayoutInflater layoutInflater;
    private int[] layouts;

    public FirstUseViewPagerAdapter(Context context) {
        layouts = new int[]{
                R.layout.slider_1_first_use,
                R.layout.slider_2_first_use,
                R.layout.slider_3_first_use,};
        layoutInflater = LayoutInflater.from(context);
    }

    @NotNull
    @Override
    public Object instantiateItem(@NotNull ViewGroup container, int position) {
        View view = layoutInflater.inflate(layouts[position], container, false);
        container.addView(view);
        return view;
    }

    @Override
    public int getCount() {
        return layouts.length;
    }

    @Override
    public boolean isViewFromObject(@NotNull View view, @NotNull Object obj) {
        return view == obj;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, @NotNull Object object) {
        View view = (View) object;
        container.removeView(view);
    }
}

File: app/src/main/java/com/bmn/bookfinder/adapters/FirstUseViewPagerAdapter.kt
package com.bmn.bookfinder.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter
import com.bmn.bookfinder.R

class FirstUseViewPagerAdapter(context: Context?) : PagerAdapter() {
    private val layoutInflater: LayoutInflater
    private val layouts: IntArray
    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = layoutInflater.inflate(layouts[position], container, false)
        container.addView(view)
        return view
    }

    override fun getCount(): Int {
        return layouts.size
    }

    override fun isViewFromObject(view: View, obj: Any): Boolean {
        return view === obj
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        val view = `object` as View
        container.removeView(view)
    }

    init {
        layouts = intArrayOf(
            R.layout.slider_1_first_use,
            R.layout.slider_2_first_use,
            R.layout.slider_3_first_use
        )
        layoutInflater = LayoutInflater.from(context)
    }
}